src/models.ts

```diff
@@ -65,6 +65,35 @@ export interface JSONRPCError {
   data?: any;
 }
 
+export class JSONRPCErrorException extends Error implements JSONRPCError {
+  public code: number;
+  public data?: any;
+
+  constructor(code: number, message: string, data?: any) {
```

Comment:
I think people typically expect the first argument of `Error` to be a `message`. How about this?
```suggestion
  constructor(message: string, code: number, data?: any) {
```

Reply:
I had it in this order to match `createJSONRPCErrorResponse`, but I have made your change since we are extending `Error`.